README.md

```diff
@@ -0,0 +1,211 @@
+  "id": 1,
+  "name": "Ubrania"
+}
+```
+
+#### Order
+
+* `/orders`
+
+GET method -> shows all orders, no values.
+
+* `/orders`
+
+POST method -> create a new order, pass values into json request body ex.:
+
+```
+{
+  "orderId": 0,
+  "userId": 0,
+  "cartId": 0,
+  "orderStatus": "CREATED"
+}
+```
+
+* `/orders/{orderId}`
+
+GET method -> returns a order with a specific id, values: orderId.
+
+* `/orders/{orderId}`
+
+PATCH method -> updates the order, pass values into json request body ex.:
+
+```
+{
+  "orderId": 0,
+  "userId": 0,
+  "cartId": 0,
+  "orderStatus": "CREATED"
+}
+```
+
+* `/orders/{orderId}`
+
+DELETE method -> deletes a order with specific id, values: orderId.
+
+
+#### Product
+
+* `/products`
+
+GET method -> shows all products, no values.
+
+* `/products/{productId}`
+
+GET method -> returns a product with a specific id, values: productId.
+
+* `/products/{productId}`
+
+DELETE method -> deletes product with specific id, values: productId.
+
+* `/products/`
+
+POST method -> create a new product, pass values into json request body ex.:
+
+```
+{
+  "id": 1,
+  "name": "kurtka zimowa"",
+  "description": "Pellentesque tempus interdum quam ut rhoncus. Donec ullamcorper turpis dolor. Donec euismod pretium eros et eleifend. Aliquam vulputate faucibus lorem non auctor. Vivamus erat turpis, molestie a nisl non, scelerisque luctus enim. Nunc mi mi, laoreet ac mollis nec, pharetra sit amet tortor. Vivamus a bibendum purus.",
+  "price": 100,
+  "groupId": 1
+}
+```
+
+* `/products/{productId}`
+
+PATCH method -> update specific values of product with given {productId} and json with new values in fields to update ex.:
+
+```
+{
+  "id": 1,
+  "name": "kurtka zimowa",
+  "description": "test",
+  "price": 300,
+  "groupId": 1
+}
+```
+
+* `/products/{productId}`
+
+PUT method -> replaces product with given of {productId} and json with new product data ex.:
+
+```
+{
+  "id": 1,
+  "name": "kurtka zimowa2",
+  "description": "test2",
+  "price": 150,
+  "groupId": 1
+}
+```
+
+#### User
+
+* `/users`
+
+GET method -> shows all users, no values.
+
+* `/users/{id}`
+
+GET method -> returns user with a specific id, values: id.
+
+* `/users/{id}`
+
+DELETE method -> deletes user with specific id, values: id.
+
+* `/users/updateUser/{id}`
+
+PUT method -> updates user with specific id, pass values into json request body.
+
+* `/users`
+
+POST method -> create a new user, pass values into json request body ex.:
+
+```
+{
+  "id": 1,
+  "username": "Piotr",
+  "status": 1,
+  "userKey": 59403,
+  "keyExpirationTime": "2023-05-18T18:42:39.827Z"
+}
+```
+
+* `/users/switchBlockade/{id}`
+
+PATCH method -> block user with specific id, values: id.
+
+* `/users/generateKey/{id}`
+
+PATCH method -> generate key(token) for user with specific id valid 1 hour, values: id.
+
+
+### What can the project be used for?
+
+This project is most simplified backend layer of generic online store. It can be developed further, for specific User requirements.
```

Comment:
```suggestion
This project is most simplified backend layer of generic online store. It can be developed further, for specific User requirements.


### Troubleshooting

It is noted, that can appear problem with testing using H2/ Gradle/ Hibernate utilities. Can be solved by addon "DB_CLOSE_ON_EXIT=FALSE;DB_CLOSE_ON_EXIT=FALSE" to url in application.properties in "test" package.

```